src/cargo/util/flock.rs

```diff
@@ -437,23 +437,58 @@ mod sys {
     use std::os::unix::io::AsRawFd;
 
     pub(super) fn lock_shared(file: &File) -> Result<()> {
-        flock(file, libc::LOCK_SH)
+        #[cfg(not(target_os = "solaris"))]
+        {
+            flock(file, libc::LOCK_SH)
+        }
+        #[cfg(target_os = "solaris")]
+        {
+            flock(file, 1)
+        }
     }
 
     pub(super) fn lock_exclusive(file: &File) -> Result<()> {
-        flock(file, libc::LOCK_EX)
+        #[cfg(not(target_os = "solaris"))]
+        {
+            flock(file, libc::LOCK_EX)
+        }
+        #[cfg(target_os = "solaris")]
+        {
+            flock(file, 2)
+        }
     }
 
     pub(super) fn try_lock_shared(file: &File) -> Result<()> {
-        flock(file, libc::LOCK_SH | libc::LOCK_NB)
+        #[cfg(not(target_os = "solaris"))]
+        {
+            flock(file, libc::LOCK_SH | libc::LOCK_NB)
+        }
+        #[cfg(target_os = "solaris")]
+        {
+            flock(file, 1 | 4)
+        }
     }
 
     pub(super) fn try_lock_exclusive(file: &File) -> Result<()> {
-        flock(file, libc::LOCK_EX | libc::LOCK_NB)
+        #[cfg(not(target_os = "solaris"))]
+        {
+            flock(file, libc::LOCK_EX | libc::LOCK_NB)
+        }
+        #[cfg(target_os = "solaris")]
+        {
+            flock(file, 2 | 4)
```

Comment:
Should we make our own constants so we limit all solaris-specific code to just the solaris implementation of `flock`

Reply:
I'm open for any suggestion. But not sure what do you exactly mean here...

Reply:
Guess something like?

```rust
#[cfg(all(unix, not(target_os = "solaris")))]
const LOCK_UN = libc::LOCK_UN;
#[cfg(all(unix, target_os = "solaris"))]
const LOCK_UN = 8;
```